netbox/netbox/api/authentication.py

```diff
@@ -123,6 +134,33 @@ def has_object_permission(self, request, view, obj):
 
         return super().has_object_permission(request, view, obj)
 
+    # Helper function to obtain the desired action from the first permission in the permission map
+    def get_action(self, method):
+        # Get the first permission
+        permission = self.perms_map.get(method)[0] if len(self.perms_map.get(method)) > 0 else None
+        if permission:
+            # Remove app and model label
+            action = permission.replace('%(app_label)s.', '').replace('_%(model_name)s', '')
+            return action
+        elif action := HTTP_ACTIONS[method]:
+            return action
+        return None
+
+
+class ViewOnlyPermissions(TokenPermissions):
```

Comment:
On the fence about the name. This might be read as "view-only," implying that a view to which it is attach does not support write operations. Maybe something like `RequireViewPermission`? I'm not sure that's any better.

Reply:
What about `RequireViewOnlyPermissions`?